File: 0011-Max-Area/0011.java
public class Solution {
  public int maxArea(int[] height) {
    int left = 0, right = height.length - 1;
    int res = 0;
    while(left < right) {
      res = Math.max(res, Math.min(height[left], height[right]) * (right - left));
      if (height[left] < height[right]) {
        left++;
      }
      else {
        right--;
      }
    }
    return res;
  }
}

File: 0011-Max-Area/0011.py
class Solution:
  def maxArea(self, height: List[int]) -> int:
    left, right = 0, len(height) - 1
    res = 0
    while left < right:
      res = max(res, min(height[left], height[right]) * (right - left))
      if height[left] <= height[right]:
        left += 1
      else:
        right -= 1
    return res